src/raft.rs

```diff
@@ -1444,6 +1459,33 @@ impl<T: Storage> Raft<T> {
         Ok(())
     }
 
+    /// Checks if this node may vote for the request. It may vote when from node
+    /// contains the same logs or more logs. When they contains same logs, priority
+    /// is considered. If priorities are still the same, and this node is also
+    /// starting campaign, then split vote happens. In this case, if `judge_split_prevote`
+    /// and `pre_vote` are enabled, it will vote only when from node has greater ID.
+    fn may_vote(&self, m: &Message) -> bool {
+        match self.raft_log.is_up_to_date(m.index, m.log_term) {
+            Ordering::Greater => true,
+            Ordering::Equal => {
+                match self.priority.cmp(&m.priority) {
+                    Ordering::Greater => false,
+                    Ordering::Equal => {
+                        // judge split vote can break symmetry of campaign, but as
+                        // it only happens during split vote, the impact should not
+                        // be significant.
```

Comment:
How can we tell if a campaign will end up split vote?

Reply:
This could cause some raft nodes with lower ID impossible to become the leader even we want to transfer leadership to it.

Reply:
If the configuration size is an odd number and leader is down, split vote can probably happen If two nodes are in PreCandidate state. `judge_split_prevote` only works on prevote, transfering leader skips prevote, so they won't have impact on the other.

Reply:
> judge_split_prevote only works on prevote

Besides transfer leader, a node needs to pass pre-campaign before start the actual campaign, so `judge_split_prevote` will impact the whole election process (pre-vote should consider enabled as this is when `judge_split_prevote` work).

Reply:
> so `judge_split_prevote` will impact the whole election process...

Indeed, it depends on whether nodes are working slowly. But in my tests, when one node is down, and after elections are finished, the leader count on each nodes don't have much differences (I remove balance leader scheduler before shutdown a node). And even it leads to more leaders on some node, it should not be a problem with the help of PD to reach a eventually balance.